tools/internal_ci/helper_scripts/prepare_build_macos_rc

```diff
@@ -67,7 +68,7 @@ then
   time git clone --depth 1 https://github.com/CocoaPods/Specs.git ~/.cocoapods/repos/master
 
   # Needed for ios-binary-size
-  time pip install --user pyyaml pyjwt==1.7.1 pyOpenSSL cryptography requests
+  time pip install --user -r $DIR/requirements.macos.txt
```

Comment:
since the "requirements.macos.txt" is only required for some jobs (AFAIK it's only needed for ios-binary-size),
should we further qualify the name of the requirements.macos.txt file  (just as you did for `requirements.linux_perf.txt`) to make it clear in what situations dependencies are actually needed?

Reply:
`prepare_build_macos_rc` is currently sourced directly for a handful of tests:

```
./tools/internal_ci/macos/grpc_distribtests.sh:23:source tools/internal_ci/helper_scripts/prepare_build_macos_rc
./tools/internal_ci/macos/grpc_interop_toprod.sh:21:source tools/internal_ci/helper_scripts/prepare_build_macos_rc
./tools/internal_ci/macos/grpc_interop_toprod.sh:22:source tools/internal_ci/helper_scripts/prepare_build_macos_interop_rc
./tools/internal_ci/macos/grpc_run_tests_matrix.sh:21:source tools/internal_ci/helper_scripts/prepare_build_macos_rc
./tools/internal_ci/macos/grpc_build_artifacts.sh:25:source tools/internal_ci/helper_scripts/prepare_build_macos_rc
./tools/internal_ci/macos/grpc_run_bazel_c_cpp_tests.sh:22:source tools/internal_ci/helper_scripts/prepare_build_macos_rc
./tools/internal_ci/macos/grpc_ios_binary_size.sh:25:source tools/internal_ci/helper_scripts/prepare_build_macos_rc
```

I haven't gone any deeper to see how these dependencies are actually used in all of these tests. If they aren't needed, we could restructure the setup scripts as a future improvement, and make these requirements specific to the binary size test.

These dependencies are only used for Obj-C and Python ... we could rename the file `requirements.macos_objc_python.txt` to be more specific, if that's meaningful here. What do you think?

Reply:
Let's leave as is. I was mostly confused about the "tabulate" and "scipy" dependencies